avalon/lib.py

```diff
@@ -281,3 +282,23 @@ def modules_from_path(path):
         modules.append(module)
 
     return modules
+
+
+def find_submodule(module, submodule):
+    """Find and return submodule of the module.
+
+    Args:
+        module (types.ModuleType): The module to search in.
+        submodule (str): The submodule name to find.
+
+    Returns:
+        types.ModuleType or None: The module, if found.
+
+    """
+    name = "%s.%s" % (module.__name__, submodule)
+    try:
+        return importlib.import_module(name)
+    except ImportError as exc:
+        if str(exc) != "No module name {}".format(name):
+            log_.warning("Could not find '%s' in module: %s" % (submodule,
+                                                                module))
```

Comment:
Just a minor thing: the preferred method for string formatting for logging functions is to pass arguments (see [here](https://docs.python.org/2/library/logging.html?#logging.Logger.debug) for more info). So I thinks it's better to write:

```python
log_.warning(
    "Could not find '%s' in module: %s",
    submodule,
    module,
)

Reply:
That's entirely true. I'll reformat hit. However I believe in many places the code is written with actual string formatting... which we should actually update too. Usually the difference is really only worth it for code that needs to be heavily optimized (runs very frequently) because the `logging` formatting in the function will only trigger when an actual log handler is connected to it and the message actually needs to be formatted. Which in our case usually is the case anyway.

Reply:
Resolved with https://github.com/getavalon/core/pull/501/commits/6d894d13133e7726d9e76bf64fec1c11985732a9